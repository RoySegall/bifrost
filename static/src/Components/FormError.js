import React from 'react';

function FormError(props) {
  const type = props['type'];
  const message = props['message'];

  if (!type) {
    return <></>;
  }

  const baseColor = type === 'error' ? 'alert-error' : 'alert-success';

  return <div className={`${baseColor} m-auto m-0 mt-4 mb-4`}
              role="alert">
    <div className="flex">
      <div className="py-1 pr-4">
        <i className="fal fa-check text-2xl"></i>
      </div>
      <div>
        <p className="text-2xl">{message}</p>
      </div>
    </div>
  </div>

  return <div className={`alert alert-${type}`} role="alert">{message}</div>
}

export default FormError;
