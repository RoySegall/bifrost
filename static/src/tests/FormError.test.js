import React from "react";
import FormError from '../Components/FormError';
import renderer from 'react-test-renderer';

test('Testing the form error component.', () => {
  const emptyFormError = renderer
    .create(<FormError />)
    .toJSON();
  expect(emptyFormError).toMatchSnapshot();

  const DangerFormError = renderer
    .create(<FormError type="danger" message="This an error" />)
    .toJSON();
  expect(DangerFormError).toMatchSnapshot();


  const SuccessFormError = renderer
    .create(<FormError type="success" message="This a success message" />)
    .toJSON();
  expect(SuccessFormError).toMatchSnapshot();
});
