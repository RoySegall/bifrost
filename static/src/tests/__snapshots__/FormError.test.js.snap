// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the form error component. 1`] = `null`;

exports[`Testing the form error component. 2`] = `
<div
  className="alert-success m-auto m-0 mt-4 mb-4"
  role="alert"
>
  <div
    className="flex"
  >
    <div
      className="py-1 pr-4"
    >
      <i
        className="fal fa-check text-2xl"
      />
    </div>
    <div>
      <p
        className="text-2xl"
      >
        This an error
      </p>
    </div>
  </div>
</div>
`;

exports[`Testing the form error component. 3`] = `
<div
  className="alert-success m-auto m-0 mt-4 mb-4"
  role="alert"
>
  <div
    className="flex"
  >
    <div
      className="py-1 pr-4"
    >
      <i
        className="fal fa-check text-2xl"
      />
    </div>
    <div>
      <p
        className="text-2xl"
      >
        This a success message
      </p>
    </div>
  </div>
</div>
`;
